.github/pull_request_template.md

```diff
@@ -0,0 +1,18 @@
+<!--- If this is a non-code change, this template is not required; reference any issues or top-level descriptions as needed -->
+<!--- All code change PRs should relate to an issue, reference it here; see example below -->
+<!--- Closes: #00000 -->
+
+Author Checklist (all must be checked):
+- [ ] Commit message includes a reference to the corresponding [GitHub issue](https://github.com/infinispan/infinispan/issues) (e.g. commit message: "[#00000] Issue title...") and is formatted according to our [git message template](https://github.com/infinispan/infinispan/blob/main/.gitmessage).
+- [ ] Commits have been squashed into self-contained units of work (e.g. 'WIP'- and 'Implements feedback'-style messages have been removed).
```

Comment:
GitHub supports "squash and merge". It avoids triggering a CI run if we manually squash it.